tests/util/transactions.ts

```diff
@@ -0,0 +1,122 @@
+import { GENESIS_ACCOUNT, GENESIS_ACCOUNT_PRIVATE_KEY } from "./constants";
+import Web3 from "web3";
+import * as RLP from "rlp";
+import { getCompiled } from "./contracts";
+import { Contract } from "web3-eth-contract";
+const debug = require("debug")("test:transaction");
+
+export interface TransactionOptions {
+  from?: string;
+  to?: string;
+  privateKey?: string;
+  nonce?: number;
+  gas?: string | number;
+  gasPrice?: string | number;
+  value?: string | number | BigInt;
+  data?: string;
+}
+
+const GENESIS_TRANSACTION: TransactionOptions = {
+  from: GENESIS_ACCOUNT,
+  privateKey: GENESIS_ACCOUNT_PRIVATE_KEY,
+  nonce: null,
+  gas: 12_000_000,
+  gasPrice: 1,
+  value: "0x00",
+};
+
+export const createTransaction = async (
+  web3: Web3,
+  options: TransactionOptions
+): Promise<string> => {
+  const gas = options.gas || 12_000_000;
+  const gasPrice = options.gasPrice !== undefined ? options.gasPrice : 1;
+  const value = options.value !== undefined ? options.value : "0x00";
+  const from = options.from || GENESIS_ACCOUNT;
+  const privateKey =
+    options.privateKey !== undefined ? options.privateKey : GENESIS_ACCOUNT_PRIVATE_KEY;
+
+  const data = {
+    from,
+    to: options.to,
+    value: value && value.toString(),
+    gasPrice,
+    gas,
+    nonce: options.nonce,
+    data: options.data,
+  };
+  debug("Sending transaction", {
+    ...data,
+    data: !data.data
+      ? ""
+      : data.data.length < 80
+      ? data.data
+      : data.data.substr(0, 7) + "..." + data.data.substr(data.data.length - 5),
+  });
```

Comment:
This is a string shortener, right? I wonder if it should be a helper function or have a comment.